src/sidebar/components/annotation-publish-control.js

```diff
@@ -15,17 +18,40 @@ import MenuItem from './menu-item';
  *
  */
 function AnnotationPublishControl({
-  group,
+  annotation,
   isDisabled,
-  isShared,
-  onCancel,
   onSave,
-  onSetPrivacy,
   settings,
 }) {
-  const publishDestination = isShared ? group.name : 'Only Me';
+  const draft = useStore(store => store.getDraft(annotation));
+  const group = useStore(store => store.getGroup(annotation.group));
+
+  const createDraft = useStore(store => store.createDraft);
+  const removeDraft = useStore(store => store.removeDraft);
+  const setDefault = useStore(store => store.setDefault);
+  const removeAnnotations = useStore(store => store.removeAnnotations);
+
+  const isPrivate = draft ? draft.isPrivate : !isShared(annotation.permissions);
+
+  const publishDestination = isPrivate ? 'Only Me' : group.name;
   const themeProps = ['ctaTextColor', 'ctaBackgroundColor'];
 
+  // Revert changes to this annotation
+  const onCancel = () => {
+    removeDraft(annotation);
+    if (isNew(annotation)) {
+      removeAnnotations([annotation]);
```

Comment:
This is functionally identical to broadcasting a `DELETE_ANNOTATION` event, as the only listener (`root_thread` service) simply wraps the `removeAnnotations` call.